### ISSUE TITLE
Gel rooms crash on steam deck

### ISSUE THREAD

**Issue by superm1** [SecondFaceSoftware/PortalRevolution-Issues#114]
**Describe the bug**
I've found somewhat randomly that rooms containing gel have tendency to hang the game on steam deck. I can still hit the steam button on the deck and pick exit.

A workaround seems to be to turn down texture detail from the default of high to "low".

**To Reproduce**

Play the game as normal and spend a while on a level with gel (6-6 perhaps?)

**Expected behavior**
No crash

**Desktop (please complete the following information):**
 - OS: steam os
 - GPU name:  custom apu
 - CPU name:  amd

**Comment by JackDotJS** [comment]
this could possibly be linux operating systems in general (or maybe just KDE Plasma? i know the steam deck has plasma in desktop mode, but idk if that has any bearing in game mode) because i am having this exact same issue. it has made the game virtually unplayable for me (at least until i found this issue with the listed workaround)

OS: Fedora 39 (DE: KDE Plasma 5.27.10)
GPU: NVIDIA GeForce RTX 4070 Ti
CPU: AMD Ryzen 7 7700X

**Comment by megaslowpoke** [comment]
I have a similar problem, but in Windows, in the first room where the gel is constantly in the room - FPS drops sharply, no tricks with video settings does not solve this, neither in the game nor global in the drivers.

But when switching settings (resolution/smoothing) or switching the water flow - something is updated in the formation of these drops of gel, and they for a while become such a curve triangular shape, and then the FPS returns to normal.

Unfortunately, this state of drops cannot be fixed in such a curve, and therefore it becomes impossible to play further. However, I have a video card much older than specified in the requirements, so I suspect that nothing can be done about it.

**Comment by Neinei0k** [comment]
Same on Linux Mint 21.2 Cinnamon. GPU: AMD Radeon RX 6600. Started hanging on 6-6. This is the first room with white gel, so I think this is what causes hanging.

**Comment by Armored-Dragon** [comment]
Same issue; orange gel that is in motion be it falling or in a tractor beam, causes crashes frequently. I have managed to brute force my way though most of the levels but one in one of the final chapters is filled with moving gel and just can not complete the test fast enough before it crashes. Levels seem to not crash anymore after the gel spawning has been disabled (where levels disable gel spawning). I think I have only noticed this issue with the speed gel. Chambers featuring only blue gel seem to be fine.

Arch Linux (Desktop, not on the Steam Deck)
Wayland session (opposed to Steam Deck's X11)
KDE Plasma Desktop environment.

**Comment by Locke** [comment]
**Describe the bug**
I've similar issue on my desktop PC.
But not a crash, just completely freezing, and I had to kill it twice.
Happens on chapter 3 (surface), in the map with the turrets, some time when the blue gel is in the beam. At the first time when it hit the second waterfall, on the second attempt when it got over the turrets.
I had no issues on the map before, with just one blue gel drop.

I might try reduced graphic settings or a different device at some other time.

**Desktop PC:**
- OS: Debian GNU/Linux 12.4 (Bookworm)
  - Gnome 43.9-0+deb12u1 (X, not Wayland)
- GPU name: NVIDIA GeForce GTX 1050 Ti/PCIe/SSE2 4GB
  - OpenGL core profile version string: 4.6.0 NVIDIA 525.147.05
- CPU name: Intel(R) Core(TM) i7 CPU 860 (overclocked)
- RAM (Quantity + Spec): 12 GB

**Comment by Organic-Marble** [comment]
This happened to me too on Debian 12. I had to turn all the video settings to Medium (they were autoset on the highest) and then I could play the game.

**Comment by tonitch** [comment]
I don't think it will help but the more input there is the better it is I guess...
I have the same issue precisely on the same level described by @Locke (chapter 3 with blue gel where you have to kill the turrets) the game freeze and I have to kill -9 to restart the game. (mean there is blocking sequence in the code as event's are not polled anymore). I managed to finish this level but then it froze again on the next level. I will wait for a patch to continue ^^

I don't think it has something to do with my os or gpu as eveyone on linux seems impacted (but if you need my info I can still provided)

I'm really pleased with the game and admire the work done! My only hope is that you don't abandon the linux gamer's because of this bug (has some did in the past). keep the good work!

**Comment by Locke** [comment]
In the meantime, I updated to 1.0.6, but the issue still happens occasionally. I have not reduced the graphic settings as suggested, just tried it again and saved often, and could continue that way, as the freeze happens only sometimes. I'm now in a much later level and occasionally have the same issue sometimes; saving often and `kill -9` is how I continue.

Like @tonitch said: the game is great, and this is the only issue I have so far.

**Comment by taisph** [comment]
Just standing in the first speed gel shower will hang the game fairly quickly on my system. Crashes appear to get worse with the speed gel in the following sections. Running under Proton (8.0-5) seems to work around the problem.

**Comment by Locke** [comment]
Great tip @taisph , I selected Proton version 8.0-5 and had no crash so far :)

**Comment by cloudrac3r** [comment]
I also had this problem on desktop Linux (Fedora 40, KDE Plasma, Flatpak Steam) especially in sp_a2_speed_gel where the level requires me to accumulate gel in mid-air between two ground portals. The game completely froze and I had to exit Steam to be able to get out of the game. This happened twice in a row. Forcing compatibility mode to Proton Experimental let me complete sp_a2_speed_gel, though the performance is a bit worse, so I'd like to be able to play native if possible.

**Comment by boogufo** [comment]
Same issue on steam deck...Crashes on any gel stage if you look at it too long. Please fix this.

**Comment by ZeeRabbit** [comment]
> Great tip @taisph , I selected Proton version 8.0-5 and had no crash so far :)

Same! I had three crashes in about 20 minutes after hitting the gel levels, and since changing to proton 8.0-5 it's been rock solid so far (steam deck)

**Comment by darkglade** [comment]
I have the same issue here.

Ubuntu 24.04.5 (With HWE for Kernel 6.14)
Intel Arc B580
Ryzen 5 2600X

First experienced in the gel cleansing level in "The Surface" with the turrets, worked for a while but now crashes pretty much when there's any substantial amount of gel on screen moving through the tractor beam.

Droping Effect Detail and Model/Texture detail to Medium let me get past that glitch.
